spec/models/reports/patient_state_spec.rb

```diff
@@ -225,6 +244,151 @@
+        patient_2 = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:encounter, patient: patient_2, encountered_on: june_2021[:now] - 1.month)
+        patient_3 = create(:patient, recorded_at: june_2021[:long_ago])
+        RefreshMaterializedViews.new.refresh_v2
+
+        with_reporting_time_zone do
+          expect(described_class.where(htn_treatment_outcome_in_last_2_months: "missed_visit", month_date: june_2021[:now]).pluck(:patient_id))
+            .to include(patient_1.id, patient_3.id)
+          expect(described_class.where(htn_treatment_outcome_in_last_2_months: "missed_visit", month_date: june_2021[:now]).pluck(:patient_id))
+            .not_to include(patient_2.id)
+        end
+      end
+      it "visited_no_bp if the patient visited, but didn't get a BP taken in the last 2 months" do
+        patient_bp_over_2_months = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:prescription_drug,
+          device_created_at: june_2021[:now] - 1.month,
+          facility: patient_bp_over_2_months.registration_facility,
+          patient: patient_bp_over_2_months,
+          user: patient_bp_over_2_months.registration_user)
+        create(:blood_pressure, patient: patient_bp_over_2_months, recorded_at: june_2021[:now] - 2.months)
+
+        patient_bp_under_2_months = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:prescription_drug,
+          device_created_at: june_2021[:now] - 1.month,
+          facility: patient_bp_under_2_months.registration_facility,
+          patient: patient_bp_under_2_months,
+          user: patient_bp_under_2_months.registration_user)
+        create(:blood_pressure, patient: patient_bp_under_2_months, recorded_at: june_2021[:now] - 1.month)
+
+        patient_with_no_bp = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:prescription_drug,
+          device_created_at: june_2021[:now] - 1.month,
+          facility: patient_with_no_bp.registration_facility,
+          patient: patient_with_no_bp,
+          user: patient_with_no_bp.registration_user)
+        RefreshMaterializedViews.new.refresh_v2
+
+        with_reporting_time_zone do
+          expect(described_class.where(htn_treatment_outcome_in_last_2_months: "visited_no_bp", month_date: june_2021[:now]).pluck(:patient_id))
+            .to include(patient_bp_over_2_months.id, patient_with_no_bp.id)
+          expect(described_class.where(htn_treatment_outcome_in_last_2_months: "visited_no_bp", month_date: june_2021[:now]).pluck(:patient_id))
+            .not_to include(patient_bp_under_2_months.id)
+        end
+      end
+      it "controlled/uncontrolled if there is a BP measured in the last 2 months that is under/not under control" do
+        patient_controlled = create(:patient, recorded_at: june_2021[:long_ago])
+        create(:blood_pressure, :with_encounter, patient: patient_controlled, recorded_at: june_2021[:now] - 1.month, systolic: 139, diastolic: 89)
```

Comment:
we can use the factory trait here, its more expressive

```suggestion
        create(:blood_pressure, :with_encounter, :under_control, patient: patient_controlled, recorded_at: june_2021[:now] - 1.month)
```

Reply:
💯  Thanks! Fixed in da2d6a8367bea0f872ed58601741ead5c2954a4b.